pkg/apis/sippyprocessing/v1/types.go

```diff
@@ -75,15 +74,11 @@ type JobRunResult struct {
 type JobResult struct {
 	Name                            string  `json:"name"`
 	Platform                        string  `json:"platform"`
-	Failures                        int     `json:"failures"`
 	KnownFailures                   int     `json:"knownFailures"`
-	Successes                       int     `json:"successes"`
-	PassPercentage                  float64 `json:"PassPercentage"`
-	PassPercentageWithKnownFailures float64 `json:"PassPercentageWithKnownFailures"`
-	TestGridUrl                     string  `json:"TestGridUrl"`
+	PassPercentageWithKnownFailures float64 `json:"passPercentageWithKnownFailures"`
+	TestGridURL                     string  `json:"TestGridURL"`
 
-	// TestResults holds entries for each test that is a part of this aggregation.  Each entry aggregates the results of all runs of a single test.  The array is sorted from lowest PassPercentage to highest PassPercentage
-	TestResults []TestResult `json:"results"`
+	SortedAggregateTestsResult `json:",inline"`
```

Comment:
i don't understand this.  testresults doesn't replace job pass percentage, they aren't the same thing.

job pass percentage:  how many times the job succeeded over how many times it failed
testresults: information about the test executionst that ran within the executions of that job

Reply:
at best this makes it very confusing when "job.TestPassPercentage" contains the "job pass percentage" (if indeed that's what you've done, i haven't looked closely) and not the testpasspercentage as aggregated across all runs of all tests in all runs of the job.

Reply:
That's not what this field should mean.  Maybe this will make it more clear https://github.com/openshift/sippy/pull/78/files#diff-d8ba90165ac3a1ed7d918c9deca1dbb9L152-L154  Those fields are dead.  We should remove them and then talk about the structure we want, which I think is a case where the aggregate links to constituent jobs and lists the success and fails for the aggregate job pass value.   This is equivalent to what you do everywhere you actually display the information.